File: Backend/src/main/java/com/ecommerce/backend/model/entity/Product.java
package com.ecommerce.backend.model.entity;


import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String itemName;

    @Column(columnDefinition = "TEXT")
    private String tags;

    private String mainImage;

    private Double rating;

    @Column(columnDefinition = "TEXT")
    private String sizes;

    @Column(columnDefinition = "TEXT")
    private String colors;

    @Column(columnDefinition = "TEXT")
    private String otherAttributes;

    @Column(columnDefinition = "TEXT")
    private String customerReviews;

    @Column(columnDefinition = "TEXT")
    private String ingredients;

    private Double initialPrice;

    @Column(columnDefinition = "TEXT")
    private String ingredientsFull;

    @Column(columnDefinition = "TEXT")
    private String categories;
}

File: FrontEnd/types/product.ts
export interface Product {
  id: number;
  itemName: string;
  description: string;
  price: number;
  quantity: number;
  category: string; // mapped from categories
  brand: string;
  sku: string;
  weight: number;
  dimensions: string;
  materials: string;
  tags: string;
  mainImage: string;
  images: string; // JSON array string
  rating: number | null;
  sizes: string; // JSON array string
  colors: string; // JSON array string
  otherAttributes: string;
  customerReviews: string;
  ingredients: string;
  initialPrice: number | null;
  ingredientsFull: string;
  categories: string;
}

export interface CartItem {
  product: Product;
  quantity: number;
}